### PR TITLE
add fused rnn cell

### PR DESCRIPTION
**Pull request by piiswrong**
@sxjscience @pluskid @sbodenstein

### CODE REVIEW

File: tests/python/gpu/test_operator_gpu.py [hunk 1]
```diff
@@ -292,15 +282,63 @@ def test_take_with_type():
...
+    mod1 = mx.mod.Module(sym1, label_names=None, context=mx.gpu(0))
+    mod1.bind(data_shapes=[('data', dshape)], label_shapes=None)
+
+    sym2, _ = cell2.unroll(5, data, merge_outputs=True)
+    mod2 = mx.mod.Module(sym2, label_names=None, context=mx.gpu(0))
+    mod2.bind(data_shapes=[('data', dshape)], label_shapes=None)
+
+    mod1.init_params()
+    args, auxs = mod1.get_params()
+    args = cell1.unpack_weights(args)
```
Comment (piiswrong) [inline tests/python/gpu/test_operator_gpu.py]: we can unpack cudnn RNN cell weight and load it with mxnet lstm symbol now @sbodenstein @pluskid @sxjscience

**Comment by sbodenstein** [comment]
@piiswrong: have you done any benchmarking comparing this fused cell to the unrolled cell? We've found that for short sequence lengths (<20) there is not that much speedup (maybe a factor of 2 at most). But for long sequence lengths, the speedup is much more significant. Do you also find this?

**Comment by sbodenstein** [comment]
@piiswrong: is there an easy way for using FusedRNN for training, but unrolled for evaluation (say on CPU) using this design? Or does the user need to copy parameters themselves to a new RNN cell after training?

File: tests/python/gpu/test_operator_gpu.py (hunk 1 above)
Comment (piiswrong) [inline tests/python/gpu/test_operator_gpu.py]: @sbodenstein See here for example of training with fused and testing with unroll

**Comment by piiswrong** [comment]
@sbodenstein The new cudnn_lstm_bucketing example shows how to train with cudnn rnn and test with mxnet symbol

**Pull request merged**